boilerplates/react-drizzle/files/components/TodoList.tsx

```diff
@@ -0,0 +1,53 @@
+import type { TodoItem } from "@batijs/drizzle/database/schema";
+import { onCreateTodo } from "@batijs/shared-telefunc/components/TodoList.telefunc";
+import { trpc } from "@batijs/trpc/trpc/client";
+import React, { useState } from "react";
+import { reload } from "vike/client/router";
+
+export function TodoList({ todoItems }: { todoItems: TodoItem[] }) {
+  const [newTodo, setNewTodo] = useState("");
+  return (
+    <>
+      <ul>
+        {todoItems.map((todoItem) => (
+          <li key={todoItem.id}>{todoItem.text}</li>
+        ))}
+        <li>
+          <form
+            onSubmit={async (ev) => {
+              ev.preventDefault();
+              if (BATI.has("telefunc")) {
+                await onCreateTodo({ text: newTodo });
+                setNewTodo("");
+                await reload();
+              } else if (BATI.has("trpc")) {
+                await trpc.onCreateTodo.mutate(newTodo);
+                setNewTodo("");
+                await reload();
+              } else {
+                try {
+                  const response = await fetch("/api/todo/create", {
+                    method: "POST",
+                    body: JSON.stringify({ text: newTodo }),
+                    headers: {
+                      "Content-Type": "application/json",
+                    },
+                  });
+                  if (response.ok) {
+                    await reload();
+                    setNewTodo("");
+                  }
+                } catch (error) {
+                  console.log("error :", error);
+                }
+              }
+            }}
```

Comment:
After thinking about this, I think we can simplify this.

First, this part is reusable for all frontend UI framework. We should move this shared code inside `boilerplates/drizzle` directly.

Second, `*-drizzle` packages like `react-drizzle` should be merged back directly to their default boilerlplates, i.e. move back `react-drizzle` content to `react`, and remove `react-drizzle`. Same goes for `vue` and `solid`.

Reply:
(I'll reference this as **B** in next comments)

Reply:
> First, this part is reusable for all frontend UI framework. We should move this shared code inside `boilerplates/drizzle` directly.

Yes, For react & solid it is very similar, but not for vue. `await reload()` doesn't work with vue, that's why I did this 
`todoItems.value.push({ id: result.lastInsertRowid as number, text: newTodo.value })`
at
https://github.com/batijs/bati/pull/258/files#diff-c3e53e4a3a9deeb2ebe0a3607a624c98feb9c25a03ccc4a72e2263ff2e12d0eaR30

Reply:
> Second, `*-drizzle` packages like `react-drizzle` should be merged back directly to their default boilerplates

Does this mean that a `TodoList` component and `pages/todo` will always be present when a user scaffolds Vike, even without using `--drizzle`?

Reply:
Yes, I plan to have todos always present in the near future anyway

Reply:
> First, this part is reusable for all frontend UI framework. We should move this shared code inside `boilerplates/drizzle` directly.

I attempted to move this part to `boilerplates/drizzle` by introducing a file named `action.ts`. However, when executing `pnpm run build`, an error occurred: 
```
  × Invalid package dependency graph: cyclic dependency detected:
  │     @batijs/trpc, @batijs/drizzle
  ╰─▶ cyclic dependency detected:
        @batijs/trpc, @batijs/drizzle
```

I suspect this issue arises from `boilerplates/trpc` having `@batijs/drizzle` in its `devDependencies`. Consequently, when we move this part to `boilerplates/drizzle`, it also requires `@batijs/trpc` in its `devDependencies`."